src/plugins/formulas/__tests__/initialization.spec.js

```diff
@@ -0,0 +1,696 @@
+      try {
+        if (this.$container2.handsontable('getInstance')) {
+          this.$container2.handsontable('getInstance').destroy();
+        }
+      } catch (e) {
+        // In some of the test cases we're manually destroying the Handsontable instances, so 'getInstance' may
+        // throw a post-mortem error.
+        if (!e.message.includes('instance has been destroyed')) {
+          throw e;
+        }
+      }
+      this.$container2.remove();
+    }
+  });
+
+  describe('Single Handsontable setup', () => {
+    it('should throw an error, when no `hyperformula` key was passed to the `formulas` settings', () => {
+      /* eslint-disable no-console */
+      const errorSpy = jasmine.createSpyObj('error', ['test']);
+      const prevError = console.error;
+
+      console.error = (...args) => {
+        errorSpy.test(...args);
+        prevError(...args);
+      };
+
+      const hot = handsontable({
+        data: getDataSimpleExampleFormulas(),
+        formulas: true,
+        licenseKey: 'non-commercial-and-evaluation'
+      });
+
+      expect(errorSpy.test).toHaveBeenCalledWith(
+        'Missing the required `hyperformula` key in the Formulas settings. ' +
+        'Please fill it with either a HyperFormula class or a HyperFormula instance.');
+      expect(hot.getPlugin('formulas').enabled).toBe(false);
+
+      console.error = prevError;
+      /* eslint-enable no-console */
+    });
+
+    it('should initialize a single working Handsontable instance and a single HyperFormula instance, when HF class' +
+      ' was passed to the `formulas` settings', () => {
+      const hot = handsontable({
+        data: getDataSimpleExampleFormulas(),
+        formulas: {
+          hyperformula: HyperFormula
+        },
+        licenseKey: 'non-commercial-and-evaluation'
+      });
+      const hfInstances = hot.getPlugin('formulas').staticRegister.getItem('hyperformulaInstances');
+      const relatedHfInstanceEntry = hfInstances.get(hot.getPlugin('formulas').hyperformula);
+
+      expect(getDataAtCell(4, 1)).toEqual(8042);
+      expect(hfInstances.size).toBeGreaterThanOrEqual(1);
+      expect(relatedHfInstanceEntry.length).toEqual(1);
+      expect(relatedHfInstanceEntry[0]).toEqual(hot.guid);
+    });
+
+    it('should initialize a single working Handsontable instance, when an external HyperFormula instance was passed' +
+      ' to the `formulas` settings', () => {
+      const hfInstance = HyperFormula.buildEmpty({ licenseKey: 'internal-use-in-handsontable' });
+      const hot = handsontable({
+        data: getDataSimpleExampleFormulas(),
+        formulas: {
+          hyperformula: hfInstance
+        },
+        licenseKey: 'non-commercial-and-evaluation'
+      });
+      const hfInstances = hot.getPlugin('formulas').staticRegister.getItem('hyperformulaInstances');
+      const relatedHfInstanceEntry = hfInstances ?
+        hfInstances.get(hot.getPlugin('formulas').hyperformula) :
+        void 0;
```

Comment:
I'm used to using `void 0`, because of a potential problem with `undefined` being overwritten by something, for example:
```
(function() {
  const undefined = 'oops';

  console.log(undefined); // 'oops'
  console.log(void 0); // undefined
})()
```

Reply:
Yeah but you could override many more things, like prototypes, that you can't really have control over. I don't want to sound negative here but why bother?

Reply:
One thing less to worry about 😄

Reply:
We should probably make it into an eslint rule at some point then, if we decide on `void 0` as our `undefined` replacement we as developers will have to worry about having to use `void 0`, or otherwise risk being inconsistent with the rest of the codebase.